torch/__init__.py

```diff
@@ -422,58 +422,80 @@ class SymInt:
     in the symbolic shape workflow.
     """
 
-    def __init__(self, node):
+    def __init__(self, node: _Any) -> None:
         # This field MUST be named node; C++ binding code assumes that this
         # class has a field named node that stores SymNode
         self.node = node
 
-    def __bool__(self):
+    def __bool__(self) -> "_bool":
         return builtins.bool(self != 0)
 
-    def __int__(self):
+    def __int__(self) -> "_int":
         return self.node.int_()
 
-    def __index__(self):
+    def __index__(self) -> "_int":
         return self.node.int_()
 
     # Magic methods installed by torch.fx.experimental.sym_node
 
-    def __round__(self, ndigits=None):
+    def __round__(self, ndigits: "_int | None" = None) -> "SymInt":
         return self
 
-    def __truediv__(self, other):
+    def __truediv__(self, other: _Union["IntLikeType", "FloatLikeType"]) -> "SymFloat":
         if isinstance(other, (builtins.float, SymFloat)):
             return sym_float(self).__float_truediv__(other)
         if not isinstance(other, (builtins.int, SymInt)):
             return NotImplemented
         return self.__int_truediv__(other)
 
-    def __rtruediv__(self, other):
+    def __rtruediv__(self, other: _Union["IntLikeType", "FloatLikeType"]) -> "SymFloat":
         if isinstance(other, (builtins.float, SymFloat)):
             return sym_float(self).__rfloat_truediv__(other)
         if not isinstance(other, (builtins.int, SymInt)):
             return NotImplemented
         return self.__rint_truediv__(other)
+    
+    @_overload
+    def __floordiv__(self, other: "IntLikeType") -> "SymInt":
+        ...
+    @_overload
+    def __floordiv__(self, other: "FloatLikeType") -> "SymFloat":
+        ...
 
-    def __floordiv__(self, other):
+    def __floordiv__(self, other: _Union["IntLikeType", "FloatLikeType"]) -> _Union["SymFloat", "SymInt"]:
         if isinstance(other, (builtins.float, SymFloat)):
             return sym_float(math.floor(sym_float(self) / other))
         if not isinstance(other, (builtins.int, SymInt)):
             return NotImplemented
         return self.__int_floordiv__(other)
+    
+    @_overload
+    def __rfloordiv__(self, other: "IntLikeType") -> "SymInt":
+        ...
+    @_overload
+    def __rfloordiv__(self, other: "FloatLikeType") -> "SymFloat":
+        ...
 
-    def __rfloordiv__(self, other):
+    # type ignore likely a bug in mypy, see https://github.com/python/mypy/issues/18498
+    def __rfloordiv__(self, other: _Union["IntLikeType", "FloatLikeType"]) -> _Union["SymFloat", "SymInt"]: # type: ignore[misc]
```

Comment:
This type ignore is somewhat ignoring, but it would appear to be a false positive from `mypy` that has not yet been resolved.